1_fetch.R

```diff
@@ -104,9 +104,10 @@ p1_targets_list <- list(
   tar_target(
     p1_site_counts,
     p1_wqp_inventory_aoi %>%
-      group_by(MonitoringLocationIdentifier) %>%
+      group_by(MonitoringLocationIdentifier, lat, lon, datum) %>%
```

Comment:
Yes! I want to retain those columns so that we can use them if needed for calculating bounding boxes. 

I've sometimes seen cases where one site identifier is used to define a site but maybe over time, slightly different lat/lon values got entered, for example. I added lat/lon/datum to `group_by()` because I wasn't sure that all unique lat/lon's would be retained using `lat = unique(lat)` if grouping by site identifier only. I just checked and it looks like these two approaches are equivalent, but I'll move `grid_id` up to the `group_by()` line to keep this code chunk smaller.

Reply:
^ Actually, in the example I give above results would be counted twice for a site identifier if lat/lon/datum/grid_id were retained in the `summarize()` step as opposed to the `group_by()` step:

```r
> tar_load(p1_wqp_inventory_aoi)
> 
> # create a small example
> inventory <- p1_wqp_inventory_aoi %>%
+   slice(1:3) %>%
+   select(MonitoringLocationIdentifier, lat, lon, datum, grid_id, resultCount) %>%
+   # add a row with the same identifier but slightly different lat/lon and result counts
+   add_row(MonitoringLocationIdentifier = p1_wqp_inventory_aoi$MonitoringLocationIdentifier[1],
+           lat = p1_wqp_inventory_aoi$lat[1] - 0.05,
+           lon = p1_wqp_inventory_aoi$lon[1] + 0.1,
+           datum = p1_wqp_inventory_aoi$datum[1],
+           resultCount = p1_wqp_inventory_aoi$resultCount[1]-20,
+           grid_id = p1_wqp_inventory_aoi$grid_id[1]) %>%
+   arrange(MonitoringLocationIdentifier)
> 
> # add all the vars we want to keep to group_by
> inventory %>%
+   group_by(MonitoringLocationIdentifier, lat, lon, datum, grid_id) %>%
+   summarize(results_count = sum(resultCount, na.rm = TRUE),
+             .groups = 'drop') %>%
+   select(MonitoringLocationIdentifier,results_count, lat, lon, datum, grid_id)
# A tibble: 4 x 6
  MonitoringLocationIdentifier results_count   lat   lon datum grid_id
  <chr>                                <dbl> <dbl> <dbl> <chr> <chr>  
1 IL_EPA-RAL-1                            11  37.6 -88.8 WGS84 11386  
2 IL_EPA-RAL-1                            31  37.6 -88.9 WGS84 11386  
3 IL_EPA-RAL-2                             5  37.6 -88.9 WGS84 11386  
4 IL_EPA-RAL-3                             5  37.6 -88.9 WGS84 11386  
> 
> # compare with summarizing those vars we want to keep
> inventory %>%
+   group_by(MonitoringLocationIdentifier) %>%
+   summarize(results_count = sum(resultCount, na.rm = TRUE),
+             lon = unique(lon),
+             lat = unique(lat),
+             datum = unique(datum),
+             grid_id = unique(grid_id)) %>%
+   select(MonitoringLocationIdentifier,results_count, lat, lon, datum, grid_id)
`summarise()` has grouped output by 'MonitoringLocationIdentifier'. You can override using the `.groups` argument.
# A tibble: 4 x 6
# Groups:   MonitoringLocationIdentifier [3]
  MonitoringLocationIdentifier results_count   lat   lon datum grid_id
  <chr>                                <dbl> <dbl> <dbl> <chr> <chr>  
1 IL_EPA-RAL-1                            42  37.6 -88.9 WGS84 11386  
2 IL_EPA-RAL-1                            42  37.6 -88.8 WGS84 11386  
3 IL_EPA-RAL-2                             5  37.6 -88.9 WGS84 11386  
4 IL_EPA-RAL-3                             5  37.6 -88.9 WGS84 11386  
> 
```

Reply:
Wow, differing lat/lons for the same site id is a scenario I have not considered before. Should we harmonize these mismatched lat/longs as part of this?

Reply:
I can't really say how often that comes up but I'm fairly confident I've seen it before. My hunch is that it's actually more common that the same lat/lon will accrue slightly different site identifiers over time, for example, as different organizations or point persons take over a monitoring site. We've definitely seen that in the DRB temperature sites.

> Should we harmonize these mismatched lat/longs as part of this?

Maybe we can include a check that looks for different lat/lon's for a given site id and just chooses the first one. I'll think more about this when we get to some light harmonization steps.